File: grpc-server/src/main/java/com/example/server/service/UserIdentifyService.java
package com.example.server.service;

import com.example.UserIdentifyGrpc.UserIdentifyImplBase;
import com.example.UserIdentifyRequest;
import com.example.UserIdentifyResponse;
import io.grpc.stub.StreamObserver;
import net.devh.boot.grpc.server.service.GrpcService;

@GrpcService
public class UserIdentifyService extends UserIdentifyImplBase {

  @Override
  public void identifyUser(UserIdentifyRequest request,
      StreamObserver<UserIdentifyResponse> responseObserver) {
    UserIdentifyResponse response = UserIdentifyResponse.newBuilder()
        .setUserCase(request.getServicePk() + "-"+request.getPlatformPk())
        .build();

    responseObserver.onNext(response);
    responseObserver.onCompleted();
  }
}

File: grpc-kotlin-lognet-server/src/main/kotlin/com/example/server/service/UserIdentifyService.kt
package com.example.server.service

import com.example.UserIdentifyGrpcKt
import com.example.UserIdentifyRequest
import com.example.UserIdentifyResponse
import kotlinx.coroutines.async
import kotlinx.coroutines.coroutineScope
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch
import org.lognet.springboot.grpc.GRpcService

@GRpcService
class UserIdentifyService: UserIdentifyGrpcKt.UserIdentifyCoroutineImplBase() {

    override suspend fun identifyUser(request: UserIdentifyRequest): UserIdentifyResponse = coroutineScope {
        launch {
            delay(100)
            println("test in launch")
        }

        val newServicePk = async {
            println("test in async")
            delay(200)
            println("test in async after delay")
            "new-service-pk"
        }

        UserIdentifyResponse.newBuilder()
            .setUserCase("${newServicePk.await()}_${request.servicePk}_${request.platformPk}")
            .build();
    }


}